src/main/java/io/github/syst3ms/skriptparser/registration/DefaultRegistration.java

```diff
@@ -42,22 +44,12 @@ public static void register() {
 
         registration.newType(Number.class,"number", "number@s")
                     .literalParser(s -> {
-                        Number n;
-                        if (s.contains(".")) {
-                            try {
-                                n = new BigDecimal(s);
-                            } catch (NumberFormatException e) {
-                                return null;
-                            }
-                        } else {
-                            try {
-                                // This allows formats like 1_000_000.
-                                n = new BigInteger(s.replaceAll("_", ""));
-                            } catch (NumberFormatException e) {
-                                return null;
-                            }
+                        if (s.matches(DECIMAL_PATTERN)) {
+                            return new BigDecimal(s);
                         }
-                        return n;
+                        // This allows formats like 1_000_000.
+                        var toMatch = s.replace("_", "");
```

Comment:
What is this for? The regex pattern does not include the ``_`` character, so this line shouldn't be necessary. The pattern works fine with or wothout this char.

Reply:
The replacement is done before the match. I added an additional check so this can't be used too freely.